openedx/core/djangoapps/xblock/api.py

```diff
@@ -208,13 +209,28 @@ def get_component_from_usage_key(usage_key: UsageKeyV2) -> Component:
     )
 
 
-def get_block_draft_olx(usage_key: UsageKeyV2) -> str:
+def get_block_olx(
+    usage_key: UsageKeyV2,
+    user: UserType,
+    *,
+    version: int | LatestVersion = LatestVersion.AUTO
+) -> str:
     """
-    Get the OLX source of the draft version of the given Learning-Core-backed XBlock.
+    Get the OLX source of the of the given Learning-Core-backed XBlock and a version.
     """
-    # Inefficient but simple approach. Optimize later if needed.
     component = get_component_from_usage_key(usage_key)
-    component_version = component.versioning.draft
+    runtime = get_runtime(user=user)
+    version = runtime.get_auto_latest_version(version)
```

Comment:
I don't want the `get_block_olx` API to accept a `user` param if it doesn't _really_ need/user the `user`. I also don't like that this initializes the runtime just to compute the version. So I think it's better to do something like:

```python
authored_data_mode = get_xblock_app_config().get_runtime_params()["authored_data_mode"]
if version == LatestVersion.AUTO:
    version = LatestVersion.DRAFT if authored_data_mode == AuthoredDataMode.DEFAULT_DRAFT else LatestVersion.PUBLISHED
```

This could be moved into a helper function.

Reply:
Updated https://github.com/openedx/edx-platform/pull/35932/commits/a66890d1cdc0c87464fbc15d5dbf8ebbd0ea38b2